test/Interop/Cxx/constructor/copy-basic.swift

```diff
@@ -0,0 +1,55 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop)
+//
+// REQUIRES: executable_test
+
+import Basic
+import StdlibUnittest
+
+var CXXCopyConstructorTestSuite = TestSuite("CXX Copy Constructor")
+
+@inline(never)
+func createHasUserProvidedCopyConstructor() -> HasUserProvidedCopyConstructor {
```

Comment:
*  Can you add a comment explaining why this needs to be a separate, `@inline(never)` function (i.e. to provoke a copy)?

*  Can you nest these functions inside the tests that use them, so that all of the code for a given test is together?

Reply:
I'm not sure these functions do really force a copy even with `@inline(never)`. The copy should be elided by a sufficiently smart frontend or optimizer.

I'd suggest to implement the test in a way that forces a copy due to semantic constraints.

Reply:
Right. I'll update these to return a tuple.